src/main/java/de/tum/cit/aet/artemis/iris/service/pyris/PyrisWebhookService.java

```diff
@@ -240,4 +245,83 @@ public Map<Long, IngestionState> getLectureUnitsIngestionState(long courseId, lo
                 .collect(Collectors.toMap(DomainObject::getId, unit -> pyrisConnectorService.getLectureUnitIngestionState(courseId, lectureId, unit.getId())));
     }
 
+    private boolean faqIngestionEnabled(Course course) {
+        return irisSettingsService.getRawIrisSettingsFor(course).getIrisFaqIngestionSettings() != null
+                && irisSettingsService.getRawIrisSettingsFor(course).getIrisFaqIngestionSettings().isEnabled();
+    }
+
+    /**
+     * send the updated / created faqs to Pyris for ingestion if autoLecturesUpdate is enabled
+     *
+     * @param courseId Id of the course where the attachment is added
+     * @param newFaq   the new faqs to be sent to pyris for ingestion
+     */
+    public void autoUpdateFaqInPyris(Long courseId, Faq newFaq) {
+        IrisCourseSettings courseSettings = irisSettingsRepository.findCourseSettings(courseId).isPresent() ? irisSettingsRepository.findCourseSettings(courseId).get() : null;
+        if (courseSettings != null && courseSettings.getIrisFaqIngestionSettings() != null && courseSettings.getIrisFaqIngestionSettings().isEnabled()
+                && courseSettings.getIrisFaqIngestionSettings().getAutoIngestOnFaqCreation()) {
+            addFaqToPyris(newFaq);
+        }
+    }
+
+    /**
+     * adds the faq to the vector database in Pyris
+     *
+     * @param faq The faq that got Updated
+     * @return jobToken if the job was created else null
+     */
+    public String addFaqToPyris(Faq faq) {
```

Comment:
just from reading the comments, it's not too easy to make out the difference between `addFaqToPyris` and `autoUpdateFaqInPyris`.

Little bit out of scope: I am aware that the JavaDoc regarding `vector database in Pyris` is used in multiple places. In my opinion, that should be irrelevant for Artemis since Artemis does not need to have any knowledge about Pyris apart from its interfaces (i.e. REST API).